packages/liquidator/src/liquidationStrategy.js

```diff
@@ -0,0 +1,254 @@
+  emit = x => x
+) => {
+  assert(liquidationDeadline >= 0, "liquidationDeadline must be 0 or higher");
+  assert(minSponsorSize, "requires minSponsorSize");
+  assert(toBN, "requires toBN");
+  assert(BN, "requires BN");
+
+  // these variables are optional and are only checked if WDF is activated
+  whaleDefenseFundWei &&
+    assert(
+      defenseActivationPercent >= 0 && defenseActivationPercent <= 100,
+      "Requires defenseActivationPercent to bet set between 0 and 100"
+    );
+  whaleDefenseFundWei && assert(liquidationLiveness > 0, "requires liquidationLiveness");
+
+  // Function which packs the arguments for a liquidation.
+  // returns parameters for empContract.methods.createLiquidation
+  function createLiquidationParams({ sponsor, tokensToLiquidate, currentBlockTime, maxCollateralPerToken }) {
+    assert(sponsor, "requires sponsor to liquidate");
+    assert(tokensToLiquidate, "requires tokenToLiquidate");
+    assert(currentBlockTime >= 0, "requires currentBlockTime");
+    assert(maxCollateralPerToken, "requires maxCollateralPerToken");
+    return [
+      sponsor,
+      { rawValue: liquidationMinPrice.toString() },
+      { rawValue: maxCollateralPerToken.toString() },
+      { rawValue: tokensToLiquidate.toString() },
+      parseInt(currentBlockTime) + liquidationDeadline
+    ];
+  }
+
+  // use in the case of a non whale defense liquidation
+  // Similar to the logic that already exists, but takes into account more information
+  function calculateTokensToLiquidate({ syntheticTokenBalance, positionTokens, maxTokensToLiquidateWei }) {
+    positionTokens = toBN(positionTokens);
+    syntheticTokenBalance = toBN(syntheticTokenBalance);
+
+    if (positionTokens.lte(0)) return toBN(0);
+
+    // available capital calculates the max we can currently spend
+    let availableCapital = BN.max(toBN(0), syntheticTokenBalance.sub(toBN(whaleDefenseFundWei)));
+    // let availableCapital = syntheticTokenBalance
+    if (maxTokensToLiquidateWei) availableCapital = BN.min(toBN(maxTokensToLiquidateWei), availableCapital);
+
+    // we have enough capital to fully liquidate positon
+    if (availableCapital.gte(positionTokens)) {
+      return positionTokens;
+    }
+
+    // we dont have enough capital for full liquidation.
+    // we check to make sure position remains above min sponsor size
+    if (positionTokens.sub(availableCapital).gte(toBN(minSponsorSize))) {
+      return availableCapital;
+    }
+
+    // our liquidation would bring it below min sponsor size, so just bring it to min
+    return BN.max(toBN("0"), positionTokens.sub(toBN(minSponsorSize)));
+  }
+
+  // Call this with only undercollateralized position
+  // SyntheticTokenBalance should be passed in respecting maxTokensToLiquidate
+  function processPosition({
+    position,
+    syntheticTokenBalance,
+    currentBlockTime,
+    maxCollateralPerToken,
+    // mandated token to liquidate amount, an override
+    maxTokensToLiquidateWei,
```

Comment:
What's the purpose of this override parameter?

Reply:
This was an existing parameter passed into `liquidatePositions` from the call in index.js. Was trying to respect this limit, but currently its just used as a proxy for the liquidators synthetic balance.